File: backend/src/main/java/com/herd/squire/models/attributes/WeaponProperty.java
package com.herd.squire.models.attributes;

public class WeaponProperty extends Attribute {
    public WeaponProperty() {}

    public WeaponProperty(String id, String name, String description, int sid, boolean author, int version) {
        super(id, name, description, AttributeType.WEAPON_PROPERTY, sid, author, version);
    }
}

File: frontend/src/app/shared/models/attributes/weapon-property.ts
import {Attribute} from './attribute';
import {AttributeType} from './attribute-type.enum';

export class WeaponProperty extends Attribute {
  type = 'WeaponProperty';

  constructor() {
    super();
    this.attributeType = AttributeType.WEAPON_PROPERTY;
  }
}
